python/tank/bootstrap/manager.py

```diff
@@ -705,44 +804,29 @@ def _cache_apps(self, tk, config_engine_name, config_entity, progress_callback,
                 # use a broader approach that will probably cache more apps,
                 # but at least the ones that we need.
                 env_name_list = pc.get_environments()
-
-            # pass 1 - populate list of all descriptors
-            for env_name in env_name_list:
-
-                env_obj = pc.get_environment(env_name, context)
-
-                for engine in env_obj.get_engines():
-
-                    # Select the descriptors for the configuration engine.
-                    if engine == config_engine_name:
-
-                        descriptors.append(env_obj.get_engine_descriptor(engine))
-
-                        for app in env_obj.get_apps(engine):
-                            descriptors.append(env_obj.get_app_descriptor(engine, app))
-
-                for framework in env_obj.get_frameworks():
-
-                    descriptors.append(env_obj.get_framework_descriptor(framework))
-
         else:
-            # Since self._caching_policy == self.CACHE_FULL, download and cache all the config dependencies.
+            env_name_list = pc.get_environments()
+            # When caching everything, ignore the config_engine_name argument.
+            config_engine_name = None
```

Comment:
I'm getting rid of that line, as I said above, I got confused between one vs all engines and one vs all environments. The doc at the top was too actually. I've fixed that.